Sources_v3/StreamChat/ChatClient.swift

```diff
@@ -305,13 +320,34 @@ public class _ChatClient<ExtraData: ExtraDataTypes> {
             builder(self.databaseContainer, self.eventNotificationCenter, self.apiClient)
         }
     }
-    
+
+    func completeConnectionIdWaiters(connectionId: String?) {
+        _connectionIdWaiters.mutate { waiters in
+            waiters.forEach { $0(connectionId) }
+            waiters.removeAll()
+        }
+    }
+
+    func completeTokenWaiters(token: Token?) {
+        _tokenWaiters.mutate { waiters in
+            waiters.forEach { $0(token) }
+            waiters.removeAll()
+        }
+    }
+
     /// Before CIS-555, `webSocketClient` was initialized in `createBackgroundWorkers` in `init` since it was passed
     /// to the background workers. Now, background workers do not access `webSocketClient` and it's not being
     /// initialized in `init`, so we initialize it here explicitly.
-    private func createWebSocketClient() {
+    func createWebSocketClient() {
         _ = webSocketClient
     }
+
+    /// Before CIS-574, `apiClient` was initialized in `createBackgroundWorkers` in `init` since it was passed
+    /// to the background workers. Now, background workers do not access `webSocketClient` and it's not being
+    /// initialized in `init`, so we initialize it here explicitly.
+    func createAPIClient() {
+        _ = apiClient
+    }
```

Comment:
PLEASE let's remove this madness here for both WS and API clients 😅

Reply:
@evsaev We need to re-think tests so creation of APIClient/WSClient is not necessary in `Client.init`

Reply:
done in cc4bab1787e818d3a2cfcfe3613f6f9f093a0e56